File: solutions/day-01/day-01.r
# advent of code
# day 1

# part 1
numbers <- scan('day-01.txt')

countIncreases <- function(numbers, partTwo = F){
  if(partTwo == F){
    sum(numbers[2:length(numbers)] > numbers[-length(numbers)])
  } else {
    sum(
      numbers[2:(length(numbers) - 2)] +
      numbers[3:(length(numbers) - 1)] +
      numbers[4:length(numbers)] >
      numbers[1:(length(numbers) - 3)] +
      numbers[2:(length(numbers) - 2)] +
      numbers[3:(length(numbers) - 1)]
    )
  }
}
countIncreases(numbers)

# part 2
countIncreases(numbers, partTwo = T)

File: solutions/day-01/day-01.py
# advent of code
# day 1

# part 1
numbers = [int(x) for x in open('day-01.txt', 'r').read().split('\n')[:-1]]

def countIncreases(numbers, partTwo = False):
  if(partTwo == False):
    print(sum([i2 > i1 for i2, i1 in zip (numbers[1:], numbers[:-1])]))
  else:
    print(sum([i2 > i1 for i2, i1 in zip (numbers[3:], numbers[:-3])]))

countIncreases(numbers)

# part 2
countIncreases(numbers, partTwo = True)
